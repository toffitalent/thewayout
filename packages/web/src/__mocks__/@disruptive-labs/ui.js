const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
