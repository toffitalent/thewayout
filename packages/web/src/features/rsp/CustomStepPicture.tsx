import { Avatar, Box, Button, Flex, Text, useWizardContext, WizardStep } from '@disruptive-labs/ui';
import { ChangeEvent, useRef, useState } from 'react';
import { MediaType } from '@two/shared';
import { API } from '@app/api';
import { useAppSelector } from '@app/hooks';
import { selectAuthUser } from '../auth';
import type { WizardRspData } from './rspData';

export const CustomStepPicture = (props: Partial<WizardStep>) => {
  const { next, data } = useWizardContext<WizardRspData>();
  const user = useAppSelector(selectAuthUser);
  const ref = useRef<HTMLInputElement | null>(null);
  const [file, setFile] = useState<File>();
  const [url, setUrl] = useState<string>(data.avatarUrl || '');

  const { nextButtonLabel } = props;

  const handleSubmit = async () => {
    let avatar;
    if (file) {
      const { key } = await API.uploads.create(MediaType.Avatar, file);
      avatar = key;
    }
    await next({
      ...data,
      ...(avatar && { avatar }),
      ...(url && { avatarUrl: url }),
    });
  };

  const handleImgChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.target.files) {
      setFile(event.target.files[0]);
      setUrl(URL.createObjectURL(event.target.files[0]));
    }
  };

  return (
    <Flex>
      <Avatar size="2xl" name={`${user?.firstName} ${user?.lastName}`} src={url} />
      <Box mt={5}>
        <Button variant="outline" mb={2} onClick={() => ref.current?.click()}>
          Edit Picture
        </Button>
        <input
          type="file"
          accept="image/jpg, image/png, image/gif"
          hidden
          ref={ref}
          onChange={handleImgChange}
        />
      </Box>
      <Text fontSize="sm" color="grey.600">
        JPG, GIF or PNG. 1MB max.
      </Text>
      <Button colorScheme="primary" fluid width={96} mt={8} onClick={handleSubmit}>
        {nextButtonLabel}
      </Button>
    </Flex>
  );
};
