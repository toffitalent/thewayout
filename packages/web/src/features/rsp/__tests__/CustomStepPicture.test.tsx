import { useWizardContext } from '@disruptive-labs/ui';
import { render } from '@test';
import { CustomStepPicture } from '../CustomStepPicture';

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('CustomStepPicture', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      setData: jest.fn(),
      goTo: jest.fn(),
      data: { avatarUrl: 'test' },
    }));
  });

  test('renders correctly', async () => {
    const { container } = render(<CustomStepPicture nextButtonLabel="Save & Next" />);

    expect(container).toMatchSnapshot();
  });
});
