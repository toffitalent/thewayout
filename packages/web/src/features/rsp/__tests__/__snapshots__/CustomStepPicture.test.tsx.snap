// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStepPicture renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="avatar s-2xl"
    >
      <div
        aria-label="Test User"
        class="initials"
      >
        TU
      </div>
    </div>
    <div
      class="marginTop-5"
    >
      <button
        class="button md outline c-primary marginBottom-2"
        type="button"
      >
        Edit Picture
      </button>
      <input
        accept="image/jpg, image/png, image/gif"
        hidden=""
        type="file"
      />
    </div>
    <p
      class="text color-grey-600 fontSize-sm"
    >
      JPG, GIF or PNG. 1MB max.
    </p>
    <button
      class="button md solid c-primary fluid width-96 marginTop-8"
      type="button"
    >
      Save & Next
    </button>
  </div>
</div>
`;
